instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/DbRequestDecorator.java

```diff
@@ -26,5 +41,58 @@ public void decorate(Span span, SdkRequest sdkRequest, ExecutionAttributes attri
     if (operation != null) {
       span.setAttribute(SemanticAttributes.DB_OPERATION, operation);
     }
+
+    DynamoDbRequest request = DynamoDbRequest.ofSdkRequest(sdkRequest);
+    if (request != null) {
+      mapRequestFields(request, sdkRequest, span);
+    }
+  }
+
+  private final void mapRequestFields(DynamoDbRequest request, SdkRequest sdkRequest, Span span) {
+    for (FieldMapping fieldMapping : request.fields()) {
+      mapRequestField(fieldMapping, sdkRequest, span);
+    }
+  }
+
+  private final void mapRequestField(FieldMapping fieldMapping, SdkRequest sdkRequest, Span span) {
+    // traverse path
+    String[] path = fieldMapping.getFields();
+    Object target = sdkRequest;
+    for (int i = 0; i < path.length && target != null; i++) {
+      target = next(target, path[i]);
+    }
+    if (target != null) {
+      String value = serialize(target);
+      if (value != null && !value.isEmpty()) {
+        span.setAttribute(fieldMapping.getAttribute(), value);
+      }
+    }
+  }
+
+  @Nullable
+  private String serialize(Object target) {
+    if (target instanceof SdkPojo) {
+      try {
+        return OBJECT_MAPPER.writeValueAsString(target);
+      } catch (JsonProcessingException e) {
+        return null;
+      }
+    }
+    if (target instanceof List) {
+      return ((List<Object>) target).stream().map(this::serialize).collect(Collectors.joining());
+    }
+    // simple type
+    return target.toString();
+  }
+
+  @Nullable
+  private Object next(Object current, String fieldName) {
```

Comment:
I believe it's possible to extract request parameters without reflection, can you check it?

https://github.com/aws/aws-xray-sdk-java/blob/master/aws-xray-recorder-sdk-aws-sdk-v2/src/main/java/com/amazonaws/xray/interceptors/TracingInterceptor.java#L148

Reply:
Yes, there is a statically typed way but only for SdkRequest. Other model classes also have appropriate method, but even though it's the same method it is not in the interface (SdkPojo). So I could propose an optimisation - direct call for first level properties and dynamic for nested (see changed class).